### PR TITLE
enhancement: Improve docs and validations

### PR DESCRIPTION
**Pull request by wvanheerde**
This PR:

- Updates the docs to include some instructions on setting up the module.
- Improves TF validations on composition, giving earlier feedback on a misconfigured composition.

### CODE REVIEW

**Review by dschrander — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -52,7 +52,27 @@ Optionally a pair of webhooks can be deployed to trigger starting or stopping an
...
+
+To setup this module requires a composition of the resources that need to be managed. Based on that input, a state machine is generated. Be aware that the composition dictates order: the resources in the composition are controlled in that order when the composition is started. The order is reversed when the composition is stopped.
+
+Please see the [examples](examples/) folder for code examples of how to implement this module.
+
+### Configuration
+
+Resource types require certain parameters in order to function. It's recommended to fill the parameters by refering to existing resources in your TF code.
+
+| Resource | Resource Type | Required Parameters |
```
Comment (marwinbaumannsbp) [inline README.md:65]: afbeelding
My OCD cannot handle the choice for () and , . Could you make this the same

File: variables.tf [hunk 2]
```diff
@@ -1,6 +1,11 @@
 variable "kms_key_arn" {
   description = "The ARN of the KMS key to use with the Lambda function"
   type        = string
+
+  validation {
+    condition     = can(regex("arn:aws:kms:[a-z0-9-]+:[0-9]+:key/[a-f0-9-]+", var.kms_key_arn))
```
Comment (marwinbaumannsbp) [inline variables.tf]: Does this validation works if the key does not exist yet? If its created together with this module? In this case only a reference is passed so i'm worried that this validation will fail in this case.

File: README.md (hunk 1 above)
Comment (wvanheerde) [reply README.md:65]: good one, fixed

File: variables.tf (hunk 2 above)
Comment (wvanheerde) [reply variables.tf]: hmm, you're right that on fresh deploys this id is only known after apply. As discussed: I will remove it for now. I can always look into improving this one later.

**Review by marwinbaumannsbp — APPROVED** [review]

**Pull request merged**